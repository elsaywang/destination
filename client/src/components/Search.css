:local(.row) {
    display: flex;
    align-items: center;
    margin-bottom: 1rem;
}

:local(.rowButtons) {
    margin-top: 25px;
}

:local(.ctas) {
    display: flex;
    align-items: end;
}

:local(.error) {
    float: right;
    margin-top: 8px;
}
