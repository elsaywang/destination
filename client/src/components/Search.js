import React, { Component } from 'react';
import { GridColumn, GridRow } from '@react/react-spectrum/Grid';
import Well from '@react/react-spectrum/Well';
import Button from '@react/react-spectrum/Button';
import Select from '@react/react-spectrum/Select';
import Switch from '@react/react-spectrum/Switch';
import NumberInput from '@react/react-spectrum/NumberInput';
import AddCircle from '@react/react-spectrum/Icon/AddCircle';
import RemoveCircle from '@react/react-spectrum/Icon/RemoveCircle';
import AdvancedSearch from './AdvancedSearch';
import CustomDateRange from './CustomDateRange';
import KeyValuePair from './KeyValuePair';
import Label from './common/Label';
import InlineErrorMessage from './common/InlineErrorMessage';
import styles from './Search.css';
import statusOptions from '../constants/signalStatusOptions';
import { isFormValid } from '../utils/searchValidation';
import { getDateRangeOptionsWithinRetentionPolicy } from '../utils/dateRangeOptions';

class Search extends Component {
    getViewRecordsOptions = () =>
        getDateRangeOptionsWithinRetentionPolicy(this.props.maxSignalRetentionDays);

    renderKVPFields = pair => {
        const {
            keyValuePairs,
            advanced,
            source,
            onRemoveClick,
            onAddClick,
            onKeyChange,
            onOperatorChange,
            onValueChange,
        } = this.props;
        const validKeyValuePairsLimit = keyValuePairs.length < 3;
        const isLastPair = id => keyValuePairs[keyValuePairs.length - 1].id === id;
        const onRemove = () => onRemoveClick(pair.id);
        const reportSuiteId = source.reportSuiteIds;

        return (
            <GridRow key={pair.id}>
                <GridColumn size={12} className={styles.keyValuePairs}>
                    <KeyValuePair
                        key={pair.id}
                        pair={pair}
                        advanced={advanced}
                        reportSuiteId={
                            reportSuiteId && reportSuiteId.length ? reportSuiteId[0] : ''
                        }
                        onKeyChange={onKeyChange}
                        onOperatorChange={onOperatorChange}
                        onValueChange={onValueChange}
                    />
                    {isLastPair(pair.id) &&
                        validKeyValuePairsLimit && (
                            <Button
                                label="Add"
                                data-test="add-button"
                                onClick={onAddClick}
                                icon={<AddCircle />}
                                variant="action"
                                quiet
                            />
                        )}
                    {pair.id !== 0 && (
                        <Button
                            label="Remove"
                            data-test="remove-button"
                            onClick={onRemove}
                            icon={<RemoveCircle />}
                            variant="action"
                            quiet
                        />
                    )}
                </GridColumn>
            </GridRow>
        );
    };

    renderCustomDatepickers = () => {
        if (!this.props.isCustomDateRangeEnabled) {
            return null;
        }

        return (
            <CustomDateRange
                customStartDate={this.props.customStartDate}
                customEndDate={this.props.customEndDate}
                onCustomStartDateChange={this.props.onCustomStartDateChange}
                onCustomEndDateChange={this.props.onCustomEndDateChange}
                minCustomStartDateDaysAgo={this.props.maxSignalRetentionDays}
            />
        );
    };

    renderCTAs = () => (
        <GridColumn className={styles.ctas}>
            <Button
                label="Clear All"
                data-test="clear-all-button"
                onClick={this.props.onClearAll}
                quiet
                variant="secondary"
            />
            <Button
                label="Search"
                data-test="search-button"
                onClick={this.props.onSearch}
                variant="cta"
                disabled={!isFormValid(this.props)}
            />
        </GridColumn>
    );

    render() {
        return (
            <GridRow>
                <GridColumn size={12}>
                    <Switch
                        className="advanced-search-toggle"
                        onChange={this.props.onAdvancedSearchChange}
                        checked={this.props.advanced}
                        data-test="advanced-search-toggle"
                        aria-label="Advanced Search"
                        label="Advanced search for Adobe Analytics"
                    />
                    <div>
                        <InlineErrorMessage
                            isInvalid={this.props.errors.reportSuites.hasError}
                            errorMessage={this.props.errors.reportSuites.errorMessage}
                        />
                    </div>
                    <Well>
                        <div data-test="search-form">
                            {this.props.advanced && (
                                <AdvancedSearch
                                    enabled={this.props.advanced}
                                    onReportSuiteChange={this.props.onReportSuiteChange}
                                    onReportSuiteSelect={value =>
                                        this.props.onReportSuiteSelect(value)
                                    }
                                    sourceName={this.props.source.name}
                                    reportSuites={this.props.reportSuites}
                                />
                            )}

                            <GridRow>
                                <GridColumn size={12}>
                                    {this.props.keyValuePairs.map(this.renderKVPFields)}

                                    <GridRow>
                                        <GridColumn size={10}>
                                            <Label value="Signal Status">
                                                <Select
                                                    className="signal-status"
                                                    data-test="signal-status"
                                                    value={this.props.signalStatus}
                                                    onChange={this.props.onSignalStatusChange}
                                                    options={statusOptions}
                                                    quiet
                                                />
                                            </Label>

                                            <Label value="View Records For">
                                                <Select
                                                    className="view-records"
                                                    data-test="view-records"
                                                    value={this.props.viewRecordsFor}
                                                    onChange={this.props.onViewRecordsChange}
                                                    options={this.getViewRecordsOptions()}
                                                    quiet
                                                />
                                            </Label>

                                            {this.renderCustomDatepickers()}

                                            <Label value="Minimum Counts">
                                                <NumberInput
                                                    className="min-counts"
                                                    data-test="min-counts"
                                                    onChange={this.props.onMinEventFiresChange}
                                                    value={this.props.minEventFires}
                                                    min={this.props.eventFiresMinimum}
                                                    step={this.props.eventFiresStep}
                                                    quiet
                                                />
                                            </Label>
                                        </GridColumn>

                                        {this.renderCTAs()}
                                    </GridRow>
                                    <InlineErrorMessage
                                        className={styles.error}
                                        isInvalid={this.props.errors.searchForm.hasError}
                                        errorMessage={this.props.errors.searchForm.errorMessage}
                                    />
                                </GridColumn>
                            </GridRow>
                        </div>
                    </Well>
                </GridColumn>
            </GridRow>
        );
    }
}

export default Search;
