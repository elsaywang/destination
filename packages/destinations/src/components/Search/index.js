import SearchWrapper from './Search';

export default SearchWrapper;
