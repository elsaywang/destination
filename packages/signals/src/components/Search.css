:local(.row) {
    display: flex;
    align-items: center;
    margin-bottom: 1rem;
}

:local(.rowButtons) {
    margin-top: 25px;
    padding-left: 120px;
}

:local(.ctas) {
    display: flex;
    align-items: end;
}

:local(.error) {
    float: right;
    margin-top: 8px;
}

:local(.fieldLabel) {
    margin-right: 20px;
}

:local(.removeButton) {
    margin-left: 0 !important;
}

@media only screen and (min-width: 768px) {
    :local(.rowButtons) {
        padding-left: 225px;
    }
}
@media only screen and (min-width: 304px) and (min-width: 768px) {
    :local(.rowButtons) {
        padding-left: 290px;
    }
}
