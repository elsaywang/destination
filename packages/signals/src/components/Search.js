import React, { Component } from 'react';
import { GridColumn, GridRow } from '@react/react-spectrum/Grid';
import Well from '@react/react-spectrum/Well';
import Button from '@react/react-spectrum/Button';
import Select from '@react/react-spectrum/Select';
import Switch from '@react/react-spectrum/Switch';
import NumberInput from '@react/react-spectrum/NumberInput';
import AddCircle from '@react/react-spectrum/Icon/AddCircle';
import RemoveCircle from '@react/react-spectrum/Icon/RemoveCircle';
import AdvancedSearch from './AdvancedSearch';
import CustomDateRange from './CustomDateRange';
import KeyValuePair from './KeyValuePair';
import FieldLabel from '@react/react-spectrum/FieldLabel';
import InlineErrorMessage from './common/InlineErrorMessage';
import styles from './Search.css';
import statusOptions from '../constants/signalStatusOptions';
import { isFormValid } from '../utils/searchValidation';
import { getDateRangeOptions } from '../utils/dateRangeOptions';

class Search extends Component {
    renderKVPFields = pair => {
        const {
            keyValuePairs,
            advanced,
            source,
            onRemoveClick,
            onAddClick,
            onKeyChange,
            onOperatorChange,
            onValueChange,
            searched,
            errors,
        } = this.props;
        const validKeyValuePairsLimit = keyValuePairs.length < 3;
        const isLastPair = id => keyValuePairs[keyValuePairs.length - 1].id === id;
        const onRemove = () => onRemoveClick(pair.id);
        const reportSuiteId = source.reportSuiteIds;

        return (
            <GridRow key={pair.id}>
                <GridColumn size={7} className={styles.row}>
                    <KeyValuePair
                        key={pair.id}
                        pair={pair}
                        advanced={advanced}
                        reportSuiteId={
                            reportSuiteId && reportSuiteId.length ? reportSuiteId[0] : ''
                        }
                        onKeyChange={onKeyChange}
                        onOperatorChange={onOperatorChange}
                        onValueChange={onValueChange}
                        searched={searched}
                        errors={errors}
                    />
                </GridColumn>
                <GridColumn size={5} className={styles.rowButtons}>
                    {isLastPair(pair.id) && validKeyValuePairsLimit && (
                        <Button
                            label="Add"
                            data-test="add-button"
                            className={styles.addButton}
                            onClick={onAddClick}
                            icon={<AddCircle />}
                            variant="action"
                            quiet
                        />
                    )}
                    {pair.id !== 0 && (
                        <Button
                            label="Remove"
                            data-test="remove-button"
                            className={styles.removeButton}
                            onClick={onRemove}
                            icon={<RemoveCircle />}
                            variant="action"
                            quiet
                        />
                    )}
                </GridColumn>
            </GridRow>
        );
    };

    renderCustomDatepickers = () => {
        if (!this.props.isCustomDateRangeEnabled) {
            return null;
        }

        return (
            <CustomDateRange
                customStartDate={this.props.customStartDate}
                customEndDate={this.props.customEndDate}
                onCustomStartDateChange={this.props.onCustomStartDateChange}
                onCustomEndDateChange={this.props.onCustomEndDateChange}
                minCustomStartDateDaysAgo={this.props.maxSignalRetentionDays}
            />
        );
    };

    renderCTAs = () => (
        <GridColumn className={styles.ctas}>
            <Button
                label="Clear All"
                data-test="clear-all-button"
                onClick={this.props.onClearAll}
                quiet
                variant="secondary"
            />
            <Button
                label="Search"
                data-test="search-button"
                onClick={this.props.onSearch}
                variant="cta"
                disabled={!isFormValid(this.props) || this.props.disabled}
            />
        </GridColumn>
    );

    render() {
        return (
            <GridRow>
                <GridColumn size={12}>
                    <Switch
                        className="advanced-search-toggle"
                        onChange={this.props.onAdvancedSearchChange}
                        checked={this.props.advanced}
                        data-test="advanced-search-toggle"
                        aria-label="Advanced Search"
                        label="Advanced search for Adobe Analytics"
                    />
                    <div>
                        <InlineErrorMessage
                            isInvalid={this.props.errors.reportSuites.hasError}
                            errorMessage={this.props.errors.reportSuites.errorMessage}
                        />
                    </div>
                    <Well>
                        <div data-test="search-form">
                            {this.props.advanced && (
                                <GridRow>
                                    <GridColumn size={12} className={styles.row}>
                                        <AdvancedSearch
                                            enabled={this.props.advanced}
                                            onReportSuiteChange={this.props.onReportSuiteChange}
                                            onReportSuiteSelect={value =>
                                                this.props.onReportSuiteSelect(value)
                                            }
                                            sourceName={this.props.source.name}
                                            reportSuites={this.props.reportSuites}
                                        />
                                    </GridColumn>
                                </GridRow>
                            )}

                            <GridRow>
                                <GridColumn size={12}>
                                    {this.props.keyValuePairs.map(this.renderKVPFields)}

                                    <GridRow>
                                        <GridColumn size={10} className={styles.row}>
                                            <FieldLabel
                                                label="Signal Status"
                                                className={styles.fieldLabel}>
                                                <Select
                                                    className="signal-status"
                                                    data-test="signal-status"
                                                    value={this.props.signalStatus}
                                                    onChange={this.props.onSignalStatusChange}
                                                    options={statusOptions}
                                                    quiet
                                                />
                                            </FieldLabel>

                                            <FieldLabel
                                                label="View Records For"
                                                className={styles.fieldLabel}>
                                                <Select
                                                    className="view-records"
                                                    data-test="view-records"
                                                    value={this.props.viewRecordsFor}
                                                    onChange={this.props.onViewRecordsChange}
                                                    options={getDateRangeOptions()}
                                                    quiet
                                                />
                                            </FieldLabel>

                                            {this.renderCustomDatepickers()}

                                            <FieldLabel label="Minimum Counts">
                                                <NumberInput
                                                    className="min-counts"
                                                    data-test="min-counts"
                                                    onChange={this.props.onMinEventFiresChange}
                                                    value={this.props.minEventFires}
                                                    min={this.props.eventFiresMinimum}
                                                    step={this.props.eventFiresStep}
                                                    quiet
                                                />
                                            </FieldLabel>
                                        </GridColumn>

                                        {this.renderCTAs()}
                                    </GridRow>
                                    <InlineErrorMessage
                                        className={styles.error}
                                        isInvalid={this.props.errors.searchForm.hasError}
                                        errorMessage={this.props.errors.searchForm.errorMessage}
                                    />
                                </GridColumn>
                            </GridRow>
                        </div>
                    </Well>
                </GridColumn>
            </GridRow>
        );
    }
}

export default Search;
