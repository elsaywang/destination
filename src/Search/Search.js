import React, { Component } from 'react';

class Search extends Component {
    constructor() {
        super();
        this.state = {
            keyValuePairs: [],
            dates: [],
        };
    }

    render() {
        return this.props.path;
    }
}

export default Search;
