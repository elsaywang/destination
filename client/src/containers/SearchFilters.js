import React, { Component } from 'react';
import Search from '../components/Search';
const initialSearchState = {
    keyValuePairs: [
        {
            id: 0,
            key: '',
            operator: '=',
            value: '',
        },
    ],
    status: 'all',
    advanced: false,
    viewRecordsFor: 7,
    minCount: 1000,
};

class SearchFilters extends Component {
    constructor() {
        super();

        this.state = initialSearchState;
    }

    onAdvancedSearchChange = value => {
        this.setState({
            advanced: value,
        });
    };

    onKeyChange = (value, event) => {
        let keyValuePairs = [...this.state.keyValuePairs];

        keyValuePairs[event.target.id].key = value;
        this.setState({ keyValuePairs });
    };

    onValueChange = (value, event) => {
        let keyValuePairs = [...this.state.keyValuePairs];

        keyValuePairs[event.target.id].value = value;
        this.setState({ keyValuePairs });
    };

    onOperatorChange = (id, value) => {
        let keyValuePairs = [...this.state.keyValuePairs];

        keyValuePairs[id].operator = value;
        this.setState({ keyValuePairs });
    };

    onAddClick = () => {
        const nextId = this.state.keyValuePairs.length;
        const newKeyValuePair = {
            id: nextId,
            key: '',
            operator: '=',
            value: '',
        };

        this.setState(prevState => ({
            keyValuePairs: [...prevState.keyValuePairs, newKeyValuePair],
        }));
    };

    onStatusChange = value => {
        this.setState({
            status: value,
        });
    };

    onViewRecordsChange = value => {
        if (value === 'custom') {
            // AAM-34805
            // invoke custom date picker
            // set state to custom date range
        } else {
            this.setState({
                viewRecordsFor: value,
            });
        }
    };

    onMinCountChange = value => {
        this.setState({
            minCount: value,
        });
    };

    onSearch = () => {
        this.props.onSearch(this.state);
    };

    onClearAll = () => {
        this.setState(initialSearchState);
    };

    render() {
        return (
            <Search
                {...this.state}
                onAdvancedSearchChange={this.onAdvancedSearchChange}
                onKeyChange={this.onKeyChange}
                onValueChange={this.onValueChange}
                onOperatorChange={this.onOperatorChange}
                onAddClick={this.onAddClick}
                onStatusChange={this.onStatusChange}
                onViewRecordsChange={this.onViewRecordsChange}
                onMinCountChange={this.onMinCountChange}
                onSearch={this.onSearch}
                onClearAll={this.onClearAll}
            />
        );
    }
}

export default SearchFilters;
