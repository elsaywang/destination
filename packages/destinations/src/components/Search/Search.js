import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { connect } from 'react-redux';
import Search from '@react/react-spectrum/Search';
import { fetchDestinations, applySearch } from '../../redux/actions/destinations';
import styles from '../../components/Layout';

class SearchWrapper extends Component {
    fetchDestinationsDebounced = _.debounce(this.props.fetchDestinations, 500);

    render() {
        const { destinations, applySearch } = this.props;
        return (
            <Search
                className={styles.search}
                id={`destinations-search`}
                placeholder="Search"
                onChange={query => {
                    applySearch(query);
                    this.fetchDestinationsDebounced();
                }}
                value={destinations.searchFormText}
            />
        );
    }
}

SearchWrapper.propTypes = {
    fetchDestinations: PropTypes.func,
};

export default connect(
    destinations => destinations,
    { fetchDestinations, applySearch },
)(SearchWrapper);
