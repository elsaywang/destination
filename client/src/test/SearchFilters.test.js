import React from 'react';
import { shallow } from 'enzyme';
import SearchFilters from '../containers/SearchFilters';
import Search from '../components/Search';

describe('<SearchFilters /> component', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<SearchFilters onSearch={mockFn} path="test" />);

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <Search /> component', () => {
            expect(wrapper.find(Search).exists()).toBe(true);
        });
    });

    describe('state changes based on events', () => {
        const initialState = {
            keyValuePairs: [
                {
                    id: 0,
                    key: '',
                    operator: '=',
                    value: '',
                },
            ],
            status: 'all',
            advanced: false,
            viewRecordsFor: 7,
            minCount: 1000,
        };

        it('.onAdvancedSearchChange() changes advanced state to given value', () => {
            expect(wrapper.state('advanced')).toBe(initialState.advanced);

            wrapper.props().onAdvancedSearchChange(true);
            expect(wrapper.state('advanced')).toBe(true);
        });

        it('.onKeyChange() changes key state to given value at given id in keyValuePairs[]', () => {
            const value = 'test';
            const event = {
                target: {
                    id: 0,
                },
            };

            expect(wrapper.state('keyValuePairs')[0].key).toBe(initialState.keyValuePairs[0].key);

            wrapper.props().onKeyChange(value, event);
            const newKeyValuePairs = [...initialState.keyValuePairs];

            newKeyValuePairs[event.target.id].key = value;

            expect(wrapper.state('keyValuePairs')[0].key).toBe(value);
        });

        it('.onValueChange() changes value state to given value at given id in keyValuePairs[]', () => {
            const value = 'test value';
            const event = {
                target: {
                    id: 0,
                },
            };

            expect(wrapper.state('keyValuePairs')[0].value).toBe(
                initialState.keyValuePairs[0].value,
            );

            wrapper.props().onValueChange(value, event);
            const newKeyValuePairs = [...initialState.keyValuePairs];

            newKeyValuePairs[event.target.id].value = value;

            expect(wrapper.state('keyValuePairs')[0].value).toBe(value);
        });

        it('.onOperatorChange() changes operator state to given value at given id in keyValuePairs[]', () => {
            const value = '>';
            const id = 0;

            expect(wrapper.state('keyValuePairs')[0].operator).toBe(
                initialState.keyValuePairs[0].operator,
            );

            wrapper.props().onOperatorChange(id, value);
            const newKeyValuePairs = [...initialState.keyValuePairs];

            newKeyValuePairs[id].operator = value;

            expect(wrapper.state('keyValuePairs')[0].operator).toBe(value);
        });

        it('.onStatusChange() changes status state to given value', () => {
            expect(wrapper.state('status')).toBe(initialState.status);

            wrapper.props().onStatusChange('unused');
            expect(wrapper.state('status')).toBe('unused');
        });

        it('.onViewRecordsChange() changes viewRecordsFor state to given value', () => {
            expect(wrapper.state('viewRecordsFor')).toBe(initialState.viewRecordsFor);

            wrapper.props().onViewRecordsChange(14);
            expect(wrapper.state('viewRecordsFor')).toBe(14);
        });

        it('.onMinCountChange() changes minCount state to given value', () => {
            expect(wrapper.state('minCount')).toBe(initialState.minCount);

            wrapper.props().onMinCountChange(10000);
            expect(wrapper.state('minCount')).toBe(10000);
        });

        xit('.onSearch() is called', () => {
            wrapper.props().onSearch(initialState);
            expect(wrapper.props().onSearch()).toHaveBeenCalled();
        });

        it('.onAddClick() adds another keyValuePair to state.keyValuePairs', () => {
            expect(wrapper.state('keyValuePairs')).toEqual(initialState.keyValuePairs);

            const expected = [
                ...initialState.keyValuePairs,
                {
                    id: initialState.keyValuePairs.length,
                    key: '',
                    operator: '=',
                    value: '',
                },
            ];

            wrapper.props().onAddClick();
            expect(wrapper.state('keyValuePairs')).toEqual(expected);
        });

        it('.onClearAll() to reset state back to initial state', () => {
            wrapper.props().onClearAll();
            expect(wrapper.state()).toEqual({
                keyValuePairs: [
                    {
                        id: 0,
                        key: '',
                        operator: '=',
                        value: '',
                    },
                ],
                status: 'all',
                advanced: false,
                viewRecordsFor: 7,
                minCount: 1000,
            });
        });
    });
});
