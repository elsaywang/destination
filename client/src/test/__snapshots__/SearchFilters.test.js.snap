// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFilters /> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchFilters
    onSearch={[MockFunction]}
    path="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "advanced": false,
      "keyValuePairs": Array [
        Object {
          "id": 0,
          "key": "",
          "operator": "=",
          "value": "",
        },
      ],
      "minCount": 1000,
      "onAddClick": [Function],
      "onAdvancedSearchChange": [Function],
      "onClearAll": [Function],
      "onKeyChange": [Function],
      "onMinCountChange": [Function],
      "onOperatorChange": [Function],
      "onSearch": [Function],
      "onStatusChange": [Function],
      "onValueChange": [Function],
      "onViewRecordsChange": [Function],
      "status": "all",
      "viewRecordsFor": 7,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "advanced": false,
        "keyValuePairs": Array [
          Object {
            "id": 0,
            "key": "",
            "operator": "=",
            "value": "",
          },
        ],
        "minCount": 1000,
        "onAddClick": [Function],
        "onAdvancedSearchChange": [Function],
        "onClearAll": [Function],
        "onKeyChange": [Function],
        "onMinCountChange": [Function],
        "onOperatorChange": [Function],
        "onSearch": [Function],
        "onStatusChange": [Function],
        "onValueChange": [Function],
        "onViewRecordsChange": [Function],
        "status": "all",
        "viewRecordsFor": 7,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
